### PR TITLE
Feature/friends UI

### PR DESCRIPTION
**Pull request by Jnf14**
This PR concerns the friends system linked directly to the Firestore DB allowing to see the Friends of a user and to search for another user.

### CODE REVIEW

File: app/src/main/java/com/ncnf/friends/ui/AddFriendFragment.java [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+import javax.inject.Inject;
+
+import dagger.hilt.android.AndroidEntryPoint;
+
+@AndroidEntryPoint
+public class AddFriendFragment extends Fragment {
+
+    private RecyclerView recycler;
+    private UserAdapter adapter;
```
Comment (HyPn0TiiK) [inline app/src/main/java/com/ncnf/friends/ui/AddFriendFragment.java:38]: Not needed as class attribute

File: app/src/main/java/com/ncnf/friends/ui/FriendsActivity.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+            if(position == 1){
+                return TAB_TITLES[position];
+            }
+            return TAB_TITLES[0];
+        }
+
+        @NonNull
+        @Override
+        public Fragment createFragment(int position) {
+            if(position == 1){
```
Comment (HyPn0TiiK) [inline app/src/main/java/com/ncnf/friends/ui/FriendsActivity.java:57]: use ternary

File: app/src/main/java/com/ncnf/friends/ui/FriendsActivity.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        TabLayout tabs = findViewById(R.id.friends_tabs);
+        new TabLayoutMediator(tabs, viewPager,
+                (tab, position) -> tab.setText(fragmentSelector.getPageTitle(position))
+        ).attach();
+    }
+
+    private static class FragmentSelector extends FragmentStateAdapter {
+
+        private final String[] TAB_TITLES = new String[]{"Friends", "Add a friend"};
+        private final int itemCount = 2;
```
Comment (HyPn0TiiK) [inline app/src/main/java/com/ncnf/friends/ui/FriendsActivity.java:39]: Make itemCount  equals to length of array

File: app/src/main/java/com/ncnf/friends/ui/FriendsFragment.java [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+import javax.inject.Inject;
+
+import dagger.hilt.android.AndroidEntryPoint;
+
+@AndroidEntryPoint
+public class FriendsFragment extends Fragment {
+
+    private RecyclerView recycler;
+    private UserAdapter adapter;
```
Comment (HyPn0TiiK) [inline app/src/main/java/com/ncnf/friends/ui/FriendsFragment.java:37]: Adapter not needed as class attribute

File: app/src/main/java/com/ncnf/friends/ui/FriendsFragment.java [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+        });
+
+        recycler.setLayoutManager(new LinearLayoutManager(getActivity()));
+        //TODO find better way to listen to optimize resources
+        adapter.startListening();
+        recycler.setAdapter(adapter);
+    }
+
+    private void displayUser(PrivateUser user){
+        Toast.makeText(getActivity(), "TEST_PROFILE_DISPLAY", Toast.LENGTH_LONG).show();
```
Comment (HyPn0TiiK) [inline app/src/main/java/com/ncnf/friends/ui/FriendsFragment.java:93]: add TODO

**Review by emmadeno — APPROVED** [review]
Great job !

**Review by HyPn0TiiK — APPROVED** [review]
Nice Job

**Pull request merged**